core/textile/buckd.go

```diff
@@ -103,15 +104,33 @@ func (tb *TextileBuckd) Start(ctx context.Context) error {
 	textile.Bootstrap()
 
 	log.Info("Welcome to bucket", fmt.Sprintf("peerID:%s", textile.HostID().String()))
+
+	// wait for the port to be listening to consider it done
+	log.Info("Waiting for 127.0.0.1 3006 to listen")
+	for {
```

Comment:
Instead of a forever loop, you can replace this with a counted for loop so as to allow some sort of timeout and we can have a sleep() for a few ms between each iteration

Reply:
This didn't work either way so went back to sleeping so at least it's not blocking users, and we can upgrade the check later.